docs/testrunner-toolkit/configuration/puppeteer.md

```diff
@@ -1,73 +1,552 @@
 
 ## Example Configuration
 
 ```yaml reference
 https://github.com/saucelabs/saucectl-puppeteer-example/blob/master/.sauce/config.yml
 ```
 
-## `puppeteer`
 
-__Description__: Details specific to the `puppeteer` project configuration.
+Each of the properties supported for running Puppeteer tests through `saucectl` is defined below.
+
+## `apiVersion`
+<p><small>| OPTIONAL | STRING |</small></p>
+
+Identifies the version of `saucectl` that is compatible with this configuration.
+
+```yaml
+apiVersion: v1alpha
+```
+---
+
+## `kind`
+<p><small>| REQUIRED | STRING/ENUM |</small></p>
+
+Specifies which framework is associated with the automation tests configured in this specification.
+
+```yaml
+kind: puppeteer
+```
+---
+
+## `defaults`
+<p><small>| OPTIONAL | OBJECT |</small></p>
+
+Specifies any default settings for the project.
+
+```yaml
+defaults:
+  - mode: docker
+```
+---
+
+### `mode`
+<p><small>| REQUIRED | STRING/ENUM |</small></p>
+
+Instructs `saucectl` run tests remotely through Sauce Labs (`sauce`) or locally on `docker`. At this time, the only supported value for this property for Puppeteer is `docker`.
+
+```yaml
+  mode: docker
+```
+---
+
+## `sauce`
+<p><small>| OPTIONAL | OBJECT |</small></p>
+
+The parent property containing all settings related to how tests are identified in the Sauce Labs platform.
+
+```yaml
+sauce:
+  region: eu-central-1
+  metadata:
+    name: Testing Puppeteer Support
+    tags:
+      - e2e
+      - release team
+      - other tag
+    build: GitHub Run $GITHUB_RUN_ID
+    concurrency: 5
+```
+---
+
+### `region`
+<p><small>| OPTIONAL | STRING/ENUM |</small></p>
+
+Specifies through which Sauce Labs data center tests will run. Valid values are: `us-west-1` or `eu-central-1`.
+
+```yaml
+  region: eu-central-1
+```
+---
+
+### `metadata`
+<p><small>| OPTIONAL | OBJECT |</small></p>
+
+The set of properties that allows you to provide additional information about your project that helps you distinguish it in the various environments in which it is used and reviewed, and also helps you apply filters to easily isolate tests based on metrics that are meaningful to you, as shown in the following example:
+
+```yaml
+metadata:
+  name: Testing Puppeteer Support
+  build: RC 10.4.a
+  tags:
+    - e2e
+    - release team
+    - beta
+    - featurex
+```
+---
+
+### `concurrency`
+<p><small>| OPTIONAL | INTEGER |</small></p>
+
+The maximum number of suites to execute concurrently. A setting of `10` runs up to 10 test suites at the same time. If the test defines more suites than that, excess suites are queued and run in order as currently running suites complete and new slots are available.
+
+```yaml
+  concurrency: 10
+```
+
+Alternatively, you can override the file setting at runtime by setting the concurrency flag as an inline parameter of the `saucectl run` command:
+
+```bash
+saucectl run --ccy 10
+```
+---
+
+## `docker`
```

Comment:
This is missing `docker only` banner.
This is also missing the `image` field explanations.

Reply:
Wasn't sure if `image` was supported for this one because there was no puppeteer image in the list of valid values and the puppeteer sample config file doesn't include it -- only `fileTransfer`. I saw you added the puppeteer image to the common file that includes the list, so I will make this update. Thanks.